openmls/src/ciphersuite/signable.rs

```diff
@@ -54,12 +54,16 @@ pub trait Signable: Sized {
         self,
         backend: &impl OpenMlsCryptoProvider,
         credential_bundle: &CredentialBundle,
-    ) -> Result<Self::SignedOutput, CredentialError>
+    ) -> Result<Self::SignedOutput, CredentialBundleError>
     where
         Self::SignedOutput: SignedStruct<Self>,
     {
-        let payload = self.unsigned_payload()?;
-        let signature = credential_bundle.sign(backend, &payload)?;
+        let payload = self
+            .unsigned_payload()
+            .map_err(|_| CredentialBundleError::LibraryError)?;
+        let signature = credential_bundle
+            .sign(backend, &payload)
+            .map_err(|_| CredentialBundleError::SigningFailed)?;
```

Comment:
If I were a consumer getting this error, I think I'd be very confused. Any chance we can be more explicit here w.r.t. why signing failed? Or are there too many potential reasons to list?